res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -592,15 +592,32 @@ TraktorS2MK3.fxHandler = function (field) {
         return;
     }
 
+    const fxButtonCount = 4;
+    const availableGroups = [ "[Channel1]", "[Channel2]" ];
+    var targetGroups = [];
     var fxNumber = parseInt(field.id[field.id.length - 1]);
-    var group = "[EffectRack1_EffectUnit" + fxNumber + "]";
 
-    // Toggle effect unit
-    TraktorS2MK3.fxButtonState[fxNumber] = !TraktorS2MK3.fxButtonState[fxNumber];
+    // Target decks whose shift button is pressed.
+    for (let group of availableGroups) {
+        if (TraktorS2MK3.shiftPressed[group]) targetGroups.push(group);
+    }
+
+    // If no shift button was pressed fall back to target all decks.
+    if (targetGroups.length === 0) targetGroups = availableGroups;
+
+    for (let group of targetGroups) {
+        TraktorS2MK3.fxButtonActive[group] = fxNumber;
+        engine.setValue(`[QuickEffectRack1_${group}]`, "loaded_chain_preset", TraktorS2MK3.fxButtonActive[group]);
+    }
 
-    engine.setValue(group, "group_[Channel1]_enable", TraktorS2MK3.fxButtonState[fxNumber]);
-    engine.setValue(group, "group_[Channel2]_enable", TraktorS2MK3.fxButtonState[fxNumber]);
-    TraktorS2MK3.outputHandler(TraktorS2MK3.fxButtonState[fxNumber], field.group, "fxButton" + fxNumber);
+    // There is no way on the controller to indicate which deck the effect applies to, but we'll keep all active buttons lit.
```

Comment:
when pressing Shift on a deck turn off Fx LEDs except the one used by the deck?

Reply:
My rationale behind keeping both lit is simply to let the user know they're working with two effects. I suppose either way is a trade-off. I'm somewhat leaning towards the way it currently is, but I'm all ears if you (or anyone else) feel otherwise

Reply:
Maybe I interpreted your comment wrog. Do you mean show only the FX light for the relevant deck while the shift button is held? If so I think that's a good idea and I'll have a look at implementimg it.